src/fides/api/ops/service/connectors/saas/connector_registry_service.py

```diff
@@ -20,65 +18,75 @@
 from fides.api.ops.schemas.connection_configuration.connection_config import (
     SaasConnectionTemplateValues,
 )
+from fides.api.ops.schemas.saas.connector_template import ConnectorTemplate
 from fides.api.ops.schemas.saas.saas_config import SaaSConfig
 from fides.api.ops.util.saas_util import (
+    encode_file_contents,
     load_config,
-    load_config_with_replacement,
-    load_dataset,
-    load_dataset_with_replacement,
+    load_config_from_string,
+    load_yaml_as_string,
+    replace_config_placeholders,
+    replace_dataset_placeholders,
 )
 
-_registry: Optional[ConnectorRegistry] = None
-registry_file = "data/saas/saas_connector_registry.toml"
 
+class ConnectorTemplateLoader(ABC):
+    @abstractmethod
+    def get_connector_templates(self) -> Dict[str, ConnectorTemplate]:
+        """Returns a map of connection templates"""
 
-class ConnectorTemplate(BaseModel):
+
+class FileConnectorTemplateLoader(ConnectorTemplateLoader):
     """
-    A collection of paths to artifacts that make up
-    a complete SaaS connector (SaaS config, dataset, etc.)
+    Loads SaaS connector templates from the data/saas directory.
     """
 
-    config: str
-    dataset: str
-    icon: str
-    human_readable: str
-
-    @validator("config")
-    def validate_config(cls, config: str) -> str:
-        """Validates the config at the given path"""
-        SaaSConfig(**load_config(config))
-        return config
+    def __init__(self) -> None:
+        self.templates: Dict[str, ConnectorTemplate] = {}
+        for file in os.listdir("data/saas/config"):
+            if file.endswith(".yml"):
+                config_file = os.path.join("data/saas/config", file)
+                config_dict = load_config(config_file)
+                connector_type = config_dict["type"]
+                human_readable = config_dict["name"]
 
-    @validator("dataset")
-    def validate_dataset(cls, dataset: str) -> str:
-        """Validates the dataset at the given path"""
-        Dataset(**load_dataset(dataset)[0])
-        return dataset
+                try:
+                    icon = encode_file_contents(f"data/saas/icon/{connector_type}.svg")
+                except FileNotFoundError:
+                    icon = encode_file_contents("data/saas/icon/default.svg")
 
-    @validator("icon")
-    def validate_icon(cls, icon: str) -> str:
-        """Validates the icon at the given path"""
-        if not exists(icon):
-            raise ValueError(f"Icon file {icon} was not found")
-        return icon
+                # store connector template for retrieval
+                try:
+                    self.templates[connector_type] = ConnectorTemplate(
+                        config=load_yaml_as_string(config_file),
+                        dataset=load_yaml_as_string(
```

Comment:
The `logger.exception` shows the full stack trace if anything goes wrong, it would include details saying the dataset was not found along with any validation failures

Reply:
ok! that seems good enough for me then, thanks for confirming that.

Reply:
although, now that codecov is complaining about it, does feel like maybe it's worth throwing in a test case here to show the errored behavior? if that's too annoying, then no problem and we can skip it...